solutions/silver/usaco-1350.mdx

```diff
@@ -0,0 +1,182 @@
+---
+id: usaco-1350
+source: USACO Silver 2023 January
+title: Bovine Acrobatics
+author: Arnav Gokhale
+---
+
+
+## Explanation
+
+Notice that it is optimal to place the heaviest cows at the bottom and the lightest cows at the top, which makes the gap as large as possible. So, we can sort the cows from heaviest to lightest. We maintain a sorted queue where the top of the queue is the current heaviest cow. If a cow can't be added to the current heaviest tower, we know it cannot be added to any tower, and therefore that cow can be ignored.
+
+However, this takes $O(c$ $logc$) operations where c is the total amount of cows, which is still too slow. To optimize this, instead of considering cows individually, we can group them up by weight and count. Then, by maintaining a queue of cow groups sorted by the weight, we can process all cows in O($N$).
```

Comment:
latex is messed up here, please see time complexity below for what it should look like